src/libstd/time.rs

```diff
@@ -712,11 +712,37 @@ mod tests {
         assert_almost_eq!(a - second + second, a);
         assert_almost_eq!(a.checked_sub(second).unwrap().checked_add(second).unwrap(), a);
 
-        // A difference of 80 and 800 years cannot fit inside a 32-bit time_t
-        if !(cfg!(unix) && crate::mem::size_of::<libc::time_t>() <= 4) {
+        // How many years fit in the system's time/duration representation?
+        cfg_if! {
+            if #[cfg(unix)] {
+                fn max_time_diff_years() -> u32 {
+                    if crate::mem::size_of::<::libc::time_t>() <= 4 {
+                        68 // approximately 2**31 seconds
+                    } else {
+                        u32::MAX
+                    }
+                }
+            } else if #[cfg(any(
+                all(target_vendor = "fortanix", target_env = "sgx"),
+                target_arch = "wasm32",
+                target_os = "cloudabi",
+            ))] {
+                fn max_time_diff_years() -> u32 {
+                    584 // approximately 2**64 nanoseconds
+                }
+            } else {
+                fn max_time_diff_years() -> u32 {
+                    u32::MAX
+                }
+            }
+        }
+
+        if max_time_diff_years() > 80 {
             let eighty_years = second * 60 * 60 * 24 * 365 * 80;
-            assert_almost_eq!(a - eighty_years + eighty_years, a);
-            assert_almost_eq!(a - (eighty_years * 10) + (eighty_years * 10), a);
+            assert_almost_eq!(a + eighty_years - eighty_years, a);
+            if max_time_diff_years > 800 {
+                assert_almost_eq!(a + (eighty_years * 10) - (eighty_years * 10), a);
```

Comment:
I'm not sure if it's worth keeping these tests considering the cfg kludge...

Reply:
Yeah this bit of the test seems pretty dubious.

Reply:
Removed